core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -82,6 +95,104 @@ public TableMetadata current() {
     return currentMetadata;
   }
 
+  @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public EncryptionManager encryption() {
```

Comment:
Can we extract most of logic here into a method of a utility class? These logic could be reused by other operations.